 ## Bài tập 2
 * Sử dụng chương trình Python trang 114 - 117 sách Hands-on Machine Learning with Scikit-Learn, Keras & TensorFlow, 2nd Edition tính lại ví dụ mô phỏng ở slides 10 thuộc Tuần 2.

 ### 1. Import numpy module

In [1]:
import numpy as np


 ### 2. Create dataset: ($x_i$,$y_i$) = (147,49), (150,53), (153,51), (160,54)

In [2]:
X = np.array([147, 150, 153, 160])
y = np.array([49, 53, 51,54])


 ### 3. Tính phương trình hồi quy:
 * Chuyển $x_i$ về dạng \begin{bmatrix}1 & 147 \\1 & 150 \\1 & 153 \\1 & 160 \end{bmatrix}

In [3]:
X_b = np.c_[np.ones((4, 1)), X]
X_b


array([[  1., 147.],
       [  1., 150.],
       [  1., 153.],
       [  1., 160.]])

 * Tính $\hat\theta$

In [4]:
theta_hat = np.linalg.inv(X_b.T.dot(X_b)).dot(X_b.T).dot(y)
theta_hat


array([5.01612903, 0.30645161])

 ### 4. Summary
 Vậy phương trình hồi quy cần tìm là: $y$ = 5.02 + 0.31$x$